tasks/sep/zora-002-fp-upgrade/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,561 @@
+
+        // Check that only bytecode diffs vs comparison contract are expected.
+        BytecodeComparison.Diff[] memory diffs = new BytecodeComparison.Diff[](19);
+        diffs[0] = BytecodeComparison.Diff({start: 1353, content: abi.encodePacked(expectedDelayedWETHProxy)});
+        diffs[1] = BytecodeComparison.Diff({start: 1423, content: abi.encodePacked(chainConfig.challenger)});
+        diffs[2] = BytecodeComparison.Diff({start: 1640, content: abi.encodePacked(expectedAnchorStateRegistryProxy)});
+        diffs[3] = BytecodeComparison.Diff({start: 2266, content: abi.encodePacked(chainConfig.proposer)});
+        diffs[4] = BytecodeComparison.Diff({start: 2742, content: abi.encodePacked(uint32(chainConfig.chainId))});
+        diffs[5] = BytecodeComparison.Diff({start: 6162, content: abi.encodePacked(expectedAnchorStateRegistryProxy)});
+        diffs[6] = BytecodeComparison.Diff({start: 6612, content: abi.encodePacked(expectedDelayedWETHProxy)});
+        diffs[7] = BytecodeComparison.Diff({start: 6882, content: abi.encodePacked(chainConfig.proposer)});
+        diffs[8] = BytecodeComparison.Diff({start: 6945, content: abi.encodePacked(chainConfig.challenger)});
+        diffs[9] = BytecodeComparison.Diff({start: 7088, content: abi.encodePacked(chainConfig.proposer)});
+        diffs[10] = BytecodeComparison.Diff({start: 8322, content: abi.encodePacked(chainConfig.proposer)});
+        diffs[11] = BytecodeComparison.Diff({start: 8385, content: abi.encodePacked(chainConfig.challenger)});
+        diffs[12] = BytecodeComparison.Diff({start: 9583, content: abi.encodePacked(uint32(chainConfig.chainId))});
+        diffs[13] = BytecodeComparison.Diff({start: 10810, content: abi.encodePacked(expectedDelayedWETHProxy)});
+        diffs[14] = BytecodeComparison.Diff({start: 13611, content: abi.encodePacked(expectedDelayedWETHProxy)});
+        diffs[15] = BytecodeComparison.Diff({start: 13958, content: abi.encodePacked(expectedAnchorStateRegistryProxy)});
+        diffs[16] = BytecodeComparison.Diff({start: 14984, content: abi.encodePacked(expectedDelayedWETHProxy)});
+        require(
+            BytecodeComparison.compare(
+                address(comparisonPermissionedDisputeGameImpl), address(expectedPermissionedDisputeGameImpl), diffs
+            ),
+            "checkPermissionedDisputeGame-100"
+        );
+    }
+
+    /// @notice Returns the implementation of a proxy.
+    /// @param _proxy The proxy to get the implementation of.
+    /// @return implementation_ The implementation of the proxy.
+    function _getImplementation(address _proxy) internal returns (address implementation_) {
+        vm.prank(address(0));
+        implementation_ = IProxy(payable(_proxy)).implementation();
+    }
+}
+
+/// @title BytecodeComparison
+/// @notice Library for comparing bytecode of two contracts.
+library BytecodeComparison {
```

Comment:
Is this new or copied from somewhere. Looks really handy. Can we pull this out into a dedicated file to facilitate reuse and write some tests for it?

Reply:
It's new, I'll move it into `src/`